crates/meilitool/src/main.rs

```diff
@@ -443,3 +467,105 @@ fn compact_index(db_path: PathBuf, index_name: &str) -> anyhow::Result<()> {
 
     bail!("Target index {index_name} not found!")
 }
+
+fn export_documents(
+    db_path: PathBuf,
+    index_name: String,
+    ignore_vectors: bool,
+    offset: Option<usize>,
+) -> anyhow::Result<()> {
+    let index_scheduler_path = db_path.join("tasks");
+    let env = unsafe { EnvOpenOptions::new().max_dbs(100).open(&index_scheduler_path) }
+        .with_context(|| format!("While trying to open {:?}", index_scheduler_path.display()))?;
+
+    let rtxn = env.read_txn()?;
+    let index_mapping: Database<Str, UuidCodec> =
+        try_opening_database(&env, &rtxn, "index-mapping")?;
+
+    for result in index_mapping.iter(&rtxn)? {
+        let (uid, uuid) = result?;
+        if uid == index_name {
+            let index_path = db_path.join("indexes").join(uuid.to_string());
+            let index =
+                Index::new(EnvOpenOptions::new(), &index_path, false).with_context(|| {
+                    format!("While trying to open the index at path {:?}", index_path.display())
+                })?;
+
+            let rtxn = index.read_txn()?;
+            let fields_ids_map = index.fields_ids_map(&rtxn)?;
+            let all_fields: Vec<_> = fields_ids_map.iter().map(|(id, _)| id).collect();
+            let embedding_configs = index.embedding_configs(&rtxn)?;
+
+            if let Some(offset) = offset {
+                eprintln!("Skipping {offset} documents");
+            }
+
+            let mut stdout = BufWriter::new(std::io::stdout());
+            for (i, ret) in index.all_documents(&rtxn)?.skip(offset.unwrap_or(0)).enumerate() {
```

Comment:
I fear this way of skipping documents from `all_documents` must be much slower to the one I proposed:

```
           let all_documents = index.documents_ids(&rtxn)?;
            if let Some(offset) = offset {
                eprintln!("Skipping {offset} documents");
            }
            for (document_index, ret) in index.iter_documents(&rtxn, all_documents.into_iter().skip(offset.unwrap_or_default()))?.enumerate() {
```

this avoids spawning each document for real. Care must be taken to add the `document_index` to `offset` though for displaying the correct progress, though